### PR TITLE
Run container as a non-root user

### PR DESCRIPTION
**Pull request by maxsokolovsky**

### CODE REVIEW

**Review by macedogm — CHANGES_REQUESTED** [review]
Thanks for making this PR!

It's only missing the line `USER 1000` right before the last instruction in the Dockefile `ENTRYPOINT ["/usr/bin/steve"]` to effectively make the container run as a non-root user.

For example, see how it's done in shell: https://github.com/rancher/shell/blob/fbde71b17730534ceb5eaed10724fb7445e2550c/package/Dockerfile#L40-L42

**Comment by macedogm** [comment]
Additionally, if possible, please, I would recommend to switch `FROM registry.suse.com/bci/bci-micro:15.5.11.2` to be `FROM registry.suse.com/bci/bci-micro:15.5`. As we don't merge bump PRs very often for the moment, using a less specific tag for the image will make it pull the last version of the base image for `bci/bci-micro:15.5`, otherwise we might end up stuck with vulnerable packages in the base OS layer due to the exact/restrict tag version in use.

**Review by macedogm — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -7,8 +7,18 @@ RUN \
...
+RUN echo "$user:x:1000:1000::/home/$user:/bin/bash" >> /etc/passwd && \
+    echo "$user:x:1000:" >> /etc/group && \
+    mkdir /home/$user && \
+    chown -R $user:$user /home/$user
+
 COPY --from=build /steve /usr/bin/steve
 # Hack to make golang do files,dns search order
 ENV LOCALDOMAIN=""
+USER $user
+WORKDIR /home/$user
```
Comment (cmurphy) [inline Dockerfile:23]: (nit) I don't think WORKDIR is needed, since there are no subsequent commands that use it as a relative path.

**Review by cmurphy — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (maxsokolovsky) [reply Dockerfile]: Fixed!

**Pull request merged**